src/utils/url.ts

```diff
@@ -11,15 +11,21 @@ export const venueInsideUrl = (venueId: string) => {
   return `/in/${venueId}`;
 };
 
-export const adminNGVenueUrl = (venueId?: string) =>
-  generatePath("/admin-ng/venue/:venueId?", {
-    venueId: venueId,
-  });
-
-export const adminNGSettigsUrl = (venueId?: string) =>
-  generatePath("/admin-ng/advanced-settings/:venueId?", {
-    venueId: venueId,
-  });
+const createAdminUrlHelperFor = (segment: string) => (
+  venueId?: string,
+  selectedTab?: string
+) =>
```

Comment:
The first param `segment` is called in the constants written here, but the second pair of params are called from the view. This can be confusing and hard to track, you can achieve the same result by using a single pair of params to improve the readability.

Instead of having shadow params like this: 
```tsx
export const adminNGVenueUrl = createAdminUrlHelperFor("venue");
```
You can have it like this: 
```tsx
export const adminNGVenueUrl = (venueId, selectedTab) => createAdminUrlHelperFor(venueId, selectedTab, "venue");
```

Reply:
It's not pairs of params on a single function.
It's a higher order function, a function that creates a helper function.

Intention can more easily be determined by looking at

```ts
const createAdminUrlHelperFor = (segment: string) => {
  if (segment) {
    return (venueId?: string, selectedTab?: string) =>
      generatePath(`/admin-ng/:segment?/:venueId?/:selectedTab?`, {
        segment: segment,
        venueId: venueId,
        selectedTab: selectedTab,
      });
  } else {
    return () => generatePath("/admin-ng");
  }
};
```

So, in that regard, the following one seems to have "shadow" params, while the first one is simply a plain old standard JS closure.
```ts
export const adminNGVenueUrl = (venueId, selectedTab) => createAdminUrlHelperFor(venueId, selectedTab, "venue");
```

Reply:
>It's not pairs of params on a single function.
It's a higher order function, a function that creates a helper function.

I know it's a HOF I was just using the `pair` terminology, to make you understand which ones I was talking about. And I know why it has `shadow` params, that's my argument. Why add this HOF complexity when you can make it a normal function?

Reply:
@denisdimitrov "Why add this HOF complexity when you can make it a normal function?" - please define normal function.

Reply:
NOTE: comments explaining what something is are provided for future reference to anyone that might need the context and are in no way reflection of the current participants' knowledge.

Reply:
Simplified in 2c33b6b8de126d7a4550fdd2ca5b20e1ad2b5168. There was no practical benefit of conflating the one root URL as an exception to the other two.